src/route.c

```diff
@@ -2277,29 +2173,17 @@ ni_rule_print(ni_stringbuf_t *out, const ni_rule_t *rule)
 	return out ? out->string : NULL;
 }
 
-void
-ni_rule_array_init(ni_rule_array_t *rules)
+ni_rule_array_t *
+ni_rule_array_new(void)
 {
-	memset(rules, 0, sizeof(*rules));
+	return xcalloc(1, sizeof(ni_rule_array_t));
```

Comment:
Just a hint: There is IMO no reason to use `xcalloc` (any more).. every result of ni_rule_array_new / ni_route_array_new is checked.